Modules/Quotas/src/Quotas.Application/Tiers/Commands/CreateQuotaForIdentity/Handler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Backbone.Modules.Quotas.Application.DTOs;
+using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Enmeshed.BuildingBlocks.Domain;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using MetricKey = Backbone.Modules.Quotas.Domain.Aggregates.Metrics.MetricKey;
+
+namespace Backbone.Modules.Quotas.Application.Tiers.Commands.CreateQuotaForIdentity;
+
+public class Handler : IRequestHandler<CreateQuotaForIdentityCommand, IdentityQuotaDefinitionDTO>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly ILogger<Handler> _logger;
+    private readonly IMetricsRepository _metricsRepository;
+
+    public Handler(IIdentitiesRepository identitiesRepository, ILogger<Handler> logger, IMetricsRepository metricsRepository)
+    {
+        _identitiesRepository = identitiesRepository;
+        _logger = logger;
+        _metricsRepository = metricsRepository;
+    }
+
+    public async Task<IdentityQuotaDefinitionDTO> Handle(CreateQuotaForIdentityCommand request, CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Handling CreateQuotaForTierCommand ...");
+
+        var identity = await _identitiesRepository.FindByAddress(request.IdentityAddress, cancellationToken, true);
+
+        var parseMetricKeyResult = MetricKey.Parse(request.MetricKey);
+
+        if (parseMetricKeyResult.IsFailure)
+            throw new DomainException(parseMetricKeyResult.Error);
+
+        var metric = await _metricsRepository.Find(parseMetricKeyResult.Value, cancellationToken);
+
+        var individualQuota = identity.CreateIndividualQuota(parseMetricKeyResult.Value, request.Max, request.Period);
+
+        await _identitiesRepository.Update(identity, cancellationToken);
+
+        _logger.LogTrace($"Successfully created assigned Quota to Identity. Identity Address: {identity.Address}");
+
+        var response = new IdentityQuotaDefinitionDTO(individualQuota.Id, new MetricDTO(metric.Key, metric.DisplayName), individualQuota.Max, individualQuota.Period);
```

Comment:
Your branch is out of sync with main. The MetricDTO constructor expects a Metric instead of key and display name.

Reply:
I'm aware, I would still change this after merging the feature branch with main, since the merge itself will be quite cumbersome due to the admin UI also.
Is that ok with you? Or would you rather I merge now?